sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -143,7 +143,7 @@ internal class TransportProducerPool : IAsyncDisposable
 
                 if (!Pool.TryGetValue(partitionId, out _) && !item.ActiveInstances.Any())
                 {
-                    return producer.CloseAsync(CancellationToken.None);
+                    return producer.CloseAsync(false, CancellationToken.None);
```

Comment:
This definitely doesn't look right to me.   I believe if we just mark the method `async` and await the result here, all that goes away.    We're explicitly not offering any synchronous paths; this pattern doesn't apply here.   If awaiting doesn't resolve the analyzer being cranky, then we'll want to talk about a suppression attribute.

Reply:
> We're explicitly not offering any synchronous paths;

Exactly, the analyzer was not complaining here.